### ISSUE TITLE
Documentation for nav mesh generation

### ISSUE THREAD

**Issue by RuiOrey** [donmccurdy/three-pathfinding#144]
Hi,
not sure if this has been addressed before but on the navmesh generation suggestion it uses blender 2.79.
Is there any suggested way to create it with a newer version of blender (without the game engine) or other software?
Thanks in advance!

**Comment by donmccurdy** [comment]
Yeah, this has become much harder since Blender removed the navmesh generation tools. I would use https://github.com/but0n/recastCLI.js but it is not as simple of a workflow.

**Comment by dirkk0** [comment]
Risking to get on everybodys nerves, I'd like to state (again) that creating a navmesh manually is easier than at least I thought and might be an option in some cases - actually in all cases I encountered.

**Comment by donmccurdy** [comment]
You're not wrong, and I'd use Blender myself here. But I also am not keen to provide user support for bugs resulting from bad topology (e.g. https://github.com/donmccurdy/three-pathfinding/issues/142) that wouldn't have been a problem with Recast.

**Comment by dirkk0** [comment]
@RuiOrey what I see in your screencast looks like a canyon; you could now create a plane that loosely resembles the ground but doesn't need all of the polys that the original ground has. Then you export this as a navmesh. Since the navmesh is invisble, no one can see the difference between the optimised navmesh and the 'real' terrain.
Does this make sense?

**Comment by donmccurdy** [comment]
The pathfinding algorithm relies on having somewhat consistent triangle shapes/sizes, and the long thin triangles shown above will lead to weird paths. That hasn't been an issue with RecastCLI as far as I can see, but I don't know what your screenshot represents ... is that Recast from Blender 2.79? I would have expected it to work as well.

**Comment by RuiOrey** [comment]
@dirkk0 is not a canyon, is something in another scale :). The screenshot represents a navmesh of an exposition area walking floor (center area) with a floor height difference (the 2 half moon shapes on the side) to the area where the exposition objects are. So basically they are flat surfaces in the real model. In fact the nav mesh seems to kind of tessalate the floor area...
About the invisibility yes when rendering the real one the navmesh is not visible.

@donmccurdy supposedly Watershed uses Recast.
<img width="448" alt="image" src="https://user-images.githubusercontent.com/53763/167216642-53689f6b-4c53-4c0b-9a49-91fec10fff50.png">

**Comment by donmccurdy** [comment]
Related:

- https://github.com/isaac-mason/recast-navigation-js

**Comment by donmccurdy** [comment]
This isn't something I'm likely to work on, but if a good resource on navmesh generation becomes available I'd be happy to link to that. :)
